js/Board.js

```diff
@@ -89,51 +89,66 @@ class Board {
     return false;
   }
 
+  // get all slots of given color
+  getSlots(color) {
+    return document.getElementsByClassName(color);
+  }
 
-  // check to see if there is a win
-  checkWin() {
-    // for each row
-    let winner = 0;
-    for (let i = this.length() - 1; i >= 0; i -= 7) {
-      // for each col, check for winners horizontally
-      for (let j = i; j > i - 7 ; j--) {
-        let count = 0;
-        let slot = board.getSlot(j);
-        let color = 'red';
+  // get row
+  getRow(slot) {
+    return parseInt(slot.parentNode.dataset.row, 10);
+  }
 
-        if (this.slotFilled(slot)) {
-          count++;
-          if (slot.classList.contains('yellow')) {
-            color = 'yellow';
-          }
-          for (let k = j - 1; k > i - 7; k--) {
-            slot = board.getSlot(k);
-            if (this.slotFilled(slot) && slot.classList.contains(color)) {
-              count++;
-            } else {
-              break;
-            }
-            if (count == 4) {
-              console.log(color + " wins!");
-              this.currentWin = true;
-              return color;
-            }
+  // get col
+  getCol(slot) {
+    return parseInt(slot.dataset.col, 10);
+  }
+
+  // find filled columns in each row that has a filled slot for given color
+  createRowMap(colorSlots) {
```

Comment:
Add a comment specifying that keys are the row numbers, maybe also that row numbers start at the top at 0